lib/ansible/modules/cloud/vmware/vmware_serial_port.py

```diff
@@ -0,0 +1,345 @@
+            - Required when I(backing=pipe)
+        type: str
+        choices:
+            - client
+            - server
+    service_uri:
+        description:
+            - Identifies the local host or a system on the network, depending on the value of I(direction)
+            - If you use the virtual machine as a server, the URI identifies the host on which the virtual machine runs. In this case, the host name part of the URI should be empty, or it should specify the address of the local host
+            - If you use the virtual machine as a client, the URI identifies the remote system on the network
+            - Required when I(backing=network)
+        type: str
+    proxy_uri:
+        description:
+            - Identifies a proxy service that provides network access to the I(serviceURI)
+            - If you specify a proxy URI, the virtual machine initiates a connection with the proxy service and forwards the I(serviceURI) and I(direction) to the proxy
+            - Required when I(backing=network)
+        type: str
+    yield_on_poll:
+        description:
+            - Enables CPU yield behavior
+        type: bool
+        default: true
+    no_rx_loss:
+        description:
+            - Enables optimized data transfer over the pipe
+            - Optional when I(backing=pipe)
+        type: bool
+    force:
+        description:
+            - Forcefully power off the VM before creating a serial port
+        type: bool
+        default: false
+
+extends_documentation_fragment:
+    - vmware.documentation
+
+author:
+    - Anusha Hegde (@anusha94)
+'''
+
+EXAMPLES = '''
+# Create serial ports
+- name: Network backing type
+  vmware_serial_port:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    name: '{{ name }}'
+    backing: network
+    service_uri: 'service_uri'
+    direction: client
+    yield_on_poll: true
+  delegate_to: localhost
+
+- name: Pipe backing type
+  vmware_serial_port:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    name: '{{ name }}'
+    backing: 'pipe'
+    pipe_name: 'pipe_name'
+    endpoint: client
+    yield_on_poll: true
+    no_rx_loss: true
+  delegate_to: localhost
+
+- name: Device backing type
+  vmware_serial_port:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    name: '{{ name }}'
+    backing: 'device'
+    device_name: 'device_name'
+    yield_on_poll: true
+  delegate_to: localhost
+
+- name: File backing type
+  vmware_serial_port:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    name: '{{ name }}'
+    backing: file
+    file_path: 'file_path'
+    yield_on_poll: true
+  delegate_to: localhost
+
+'''
+
+RETURN = r'''
+#
```

Comment:
Please add return output (This gives user a better idea about expected output and format).